p2p/libp2p/netMessenger.go

```diff
@@ -686,6 +687,46 @@ func (netMes *networkMessenger) Bootstrap() error {
 	return err
 }
 
+// WaitForConnections will wait the maxWaitingTime duration or until the target connected peers was achieved
+func (netMes *networkMessenger) WaitForConnections(maxWaitingTime time.Duration, minNumOfPeers uint32) {
+	startTime := time.Now()
+	defer func() {
+		log.Debug("networkMessenger.WaitForConnections",
+			"waited", time.Since(startTime), "num connected peers", len(netMes.ConnectedPeers()))
+	}()
+
+	if minNumOfPeers <= 0 {
+		time.Sleep(maxWaitingTime)
+		log.Debug("networkMessenger.WaitForConnections", "waiting", maxWaitingTime)
```

Comment:
Perhaps change the order of these two lines?
Right now you `Sleep(maxWaitingTime)` and upon finishing waiting you add a message in logger that you are waiting, when in fact you actually finished waiting

Reply:
done